packages/core/contracts/oracle/implementation/ResultComputationV2.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+/**
+ * @title Computes vote results.
+ * @dev The result is the mode of the added votes. Otherwise, the vote is unresolved.
+ */
+library ResultComputationV2 {
+    /****************************************
+     *   INTERNAL LIBRARY DATA STRUCTURE    *
+     ****************************************/
+
+    struct Data {
+        // Maps price to number of tokens that voted for that price.
+        mapping(int256 => uint256) voteFrequency;
+        // The total votes that have been added.
+        uint256 totalVotes;
+        // The price that is the current mode, i.e., the price with the highest frequency in `voteFrequency`.
+        int256 currentMode;
+    }
+
+    /****************************************
+     *            VOTING FUNCTIONS          *
+     ****************************************/
+
+    /**
+     * @notice Adds a new vote to be used when computing the result.
+     * @param data contains information to which the vote is applied.
+     * @param votePrice value specified in the vote for the given `numberTokens`.
+     * @param numberTokens number of tokens that voted on the `votePrice`.
+     */
+    function addVote(
+        Data storage data,
+        int256 votePrice,
+        uint256 numberTokens
+    ) internal {
+        data.totalVotes += numberTokens;
+        data.voteFrequency[votePrice] += numberTokens;
+        if (votePrice != data.currentMode && data.voteFrequency[votePrice] > data.voteFrequency[data.currentMode])
+            data.currentMode = votePrice;
+    }
+
+    /****************************************
+     *        VOTING STATE GETTERS          *
+     ****************************************/
+
+    /**
+     * @notice Returns whether the result is resolved, and if so, what value it resolved to.
+     * @dev `price` should be ignored if `isResolved` is false.
+     * @param data contains information against which the `minVoteThreshold` is applied.
+     * @param minVoteThreshold min (exclusive) number of tokens that must have voted for the result to be valid. Can be
+     * used to enforce a minimum voter participation rate, regardless of how the votes are distributed.
+     * @return isResolved indicates if the price has been resolved correctly.
+     * @return price the price that the dvm resolved to.
+     */
+    function getResolvedPrice(Data storage data, uint256 minVoteThreshold)
+        internal
+        view
+        returns (bool isResolved, int256 price)
+    {
+        uint256 modeThreshold = 5e17;
```

Comment:
Unrelated to this PR, but do you think we should make this 5e17 + 1?

This would eliminate the (very rare) situation where there's a tie and the voters who finish revealing first win instead of the vote being rolled.